library/ssl_misc.h

```diff
@@ -539,11 +536,12 @@ struct mbedtls_ssl_handshake_params
 {
     /* Frequently-used boolean or byte fields (placed early to take
      * advantage of smaller code size for indirect access on Arm Thumb) */
-    uint8_t max_major_ver;              /*!< max. major version client*/
-    uint8_t max_minor_ver;              /*!< max. minor version client*/
     uint8_t resume;                     /*!<  session resume indicator*/
     uint8_t cli_exts;                   /*!< client extension presence*/
 
+    /*!< Minimum minor version to be negotiated. */
```

Comment:
I think the logic about what versions are negotiated is good and correct, but it took me some time to understand, so perhaps some more documentation would help. I'm not sure what the right places would be, but here's a suggestion:
- here, explain briefly why we need this in addition to the field with the same name in `ssl_config`
- where `ssl_context::minor_ver` is declared, mention that it's time-multiplexed: at the beginning it stored the maximum version we're willing to negotiate, and after the negotiation it stores the negotiated version.

Oh well, writing this I realise that there's already a comment about `ssl_context::minor_ver` and `ssl_session::minor_ver` being currently duplicated and potentially de-duplicated once negotiation is implemented. I'm OK with leaving that out of this PR and coming back to it later if we have an issue to track it.